### PR TITLE
Fix environ for macOS

### PR DESCRIPTION
**Pull request by barracuda156**
Fixes linking error:
```
Undefined symbols:
  "_environ", referenced from:
      _environ$non_lazy_ptr in gprocess.c.o
     (maybe you meant: _environ$non_lazy_ptr)
ld: symbol(s) not found
collect2: error: ld returned 1 exit status
```
We can just enable `environ` on macOS with an appropriate declaration.

P. S. I think original macros `SYSLOG_NG_HAVE_ENVIRON` failed to have intended effect, because `#ifdef` were used, but `#if` should have been. But anyway, rather just remove this unnecessary tweaks.

### CODE REVIEW

**Comment by kira-syslogng** [comment]
This user does not have permission to start the build. Can one of the admins verify this patch and start the build?
(admin: you have the next options (make sure you checked the code):
"ok to test" to accept this pull request (and further changes) for testing
"test this please" for a one time test run
"add to whitelist" add author of a Pull Request to whitelist (globally, be careful, it means this user can trigger kira for any PR)
do nothing -> CI won't start)

**Comment by kira-syslogng** [comment]
This user does not have permission to start the build. Can one of the admins verify this patch and start the build?
(admin: you have the next options (make sure you checked the code):
"ok to test" to accept this pull request (and further changes) for testing
"test this please" for a one time test run
"add to whitelist" add author of a Pull Request to whitelist (globally, be careful, it means this user can trigger kira for any PR)
do nothing -> CI won't start)

File: CMakeLists.txt [hunk 1]
```diff
@@ -144,10 +144,6 @@ if (HAVE_TCP_KEEPIDLE AND HAVE_TCP_KEEPINTVL AND HAVE_TCP_KEEPCNT)
   set(SYSLOG_NG_HAVE_TCP_KEEPALIVE_TIMERS 1)
 endif()
 
-if (NOT APPLE)
-  set(SYSLOG_NG_HAVE_ENVIRON 1)
-endif()
-
```
Comment (HofiOne) [inline CMakeLists.txt:150]: please remove the according block from the automake configure file `configure.ac` as well

**Review by HofiOne — CHANGES_REQUESTED** [review]

File: CMakeLists.txt (hunk 1 above)
Comment (barracuda156) [reply CMakeLists.txt:150]: Done

**Review by HofiOne — APPROVED** [review]

**Comment by barracuda156** [comment]
Thanks for merging!

**Comment by HofiOne** [comment]
Thanks for contributing!

**Pull request merged**